plugin/storage/es/spanstore/dbmodel/from_domain.go

```diff
@@ -120,9 +120,9 @@ func (fd FromDomain) convertLogs(logs []model.Log) []Log {
 }
 
 func (fd FromDomain) convertProcess(process *model.Process) Process {
-	tags, tagsMap := fd.convertKeyValuesString(process.Tags)
+	tags, tagsMap := fd.convertKeyValuesString(process.GetTags())
 	return Process{
-		ServiceName: process.ServiceName,
+		ServiceName: process.GetServiceName(),
```

Comment:
So this is exactly the type of change that I did not want to make. It avoids a panic, but does it silently, creating invalid data in the process (service name is not allowed to be empty, it could break many other places, including the UI).

I've added a sanitizer in #3631. We can try invoking it early in the Kafka consumer path.

Reply:
Thanks for taking a look at this and making things clearer. Let me take a look at the sanitizer and try what you suggested.

It make sense that the service name must not be empty, so I might revert the change to process.GetServiceName() here. For the process.GetTags, I think it still nicely handles the `nil` cases those led to the panics.

Reply:
> I think it still nicely handles the nil cases those led to the panics.

Yes, but again, it does that accidentally, it so happens that in this context the behavior is appropriate. But fundamentally you're trying to work around malformed data that should've never gotten to this point in the pipeline. *I don't mind this specific change*, but at least with the current code we knew that there is a data issue (even though panicing on it is not ideal reaction).

Reply:
Hi @yurishkuro , thank you so much for your review. As you mentioned, I understand it's important to detect data issue in our system, so should we return error here and let caller know that we do not accept the nil process for example? 

I think panic should be considered if we do want application immediately crash. Otherwise, we can consider error.

Thank you for your explanation and effort.

Reply:
If we apply the sanitizer earlier in the pipeline then this code does need to change as it will never have to deal with nil process.